x-pack/plugins/log_explorer/public/components/common/popover_chip.tsx

```diff
@@ -0,0 +1,145 @@
+import {
+  EuiBadge,
+  type EuiBadgeProps,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiIcon,
+  EuiPopover,
+  useEuiFontSize,
+  EuiPopoverFooter,
+  EuiText,
+  EuiButtonIcon,
+} from '@elastic/eui';
+import { css } from '@emotion/react';
+import { closeCellActionPopoverText, openCellActionPopoverAriaText } from './translations';
+import { FilterInButton } from './filter_in_button';
+import { FilterOutButton } from './filter_out_button';
+import { CopyButton } from './copy_button';
+
+interface ChipWithPopoverProps {
+  /**
+   * ECS mapping for the key
+   */
+  property: string;
+  /**
+   * Value for the mapping, which will be displayed
+   */
+  text: string;
+  dataTestSubj?: string;
+  leftSideIcon?: EuiBadgeProps['iconType'];
+  rightSideIcon?: EuiBadgeProps['iconType'];
+  borderColor?: string | null;
+  style?: React.CSSProperties;
+  shouldRenderPopover?: boolean;
+}
+
+const convertStyleToString = (style: React.CSSProperties) => {
+  return Object.entries(style)
+    .map(([key, value]) => `${key}: ${value};`)
+    .join(' ');
+};
+
+export function ChipWithPopover({
+  property,
+  text,
+  dataTestSubj = `${property}_${text.replace(/ +/g, '')}`,
+  leftSideIcon,
+  rightSideIcon,
+  borderColor,
+  style,
+  shouldRenderPopover = true,
+}: ChipWithPopoverProps) {
+  const xsFontSize = useEuiFontSize('xs').fontSize;
+  const [isPopoverOpen, setIsPopoverOpen] = useState(false);
+
+  const handleChipClick = useCallback(() => {
+    if (!shouldRenderPopover) return;
+    setIsPopoverOpen(true);
+  }, [shouldRenderPopover]);
+
+  const closePopover = () => setIsPopoverOpen(false);
+
+  const chipContent = (
+    <EuiBadge
+      color="hollow"
+      iconType={rightSideIcon}
+      iconSide="right"
+      data-test-subj={`dataTablePopoverChip_${dataTestSubj}`}
+      onClick={handleChipClick}
+      onClickAriaLabel={openCellActionPopoverAriaText}
+      css={css`
+        ${borderColor ? `border: 2px solid ${borderColor};` : ''}
+        font-size: ${xsFontSize};
+        display: flex;
+        justify-content: center;
+        ${style ? css(convertStyleToString(style)) : ''}
+      `}
+    >
+      <EuiFlexGroup gutterSize="xs">
+        {leftSideIcon && (
+          <EuiFlexItem>
+            <EuiIcon type={leftSideIcon} />
+          </EuiFlexItem>
+        )}
+        <EuiFlexItem>{text}</EuiFlexItem>
+      </EuiFlexGroup>
+    </EuiBadge>
+  );
+
+  return (
+    <EuiPopover
+      button={chipContent}
+      isOpen={isPopoverOpen}
+      closePopover={closePopover}
+      anchorPosition="downCenter"
+      panelPaddingSize="s"
+      panelStyle={{ minWidth: '24px' }}
+    >
+      <EuiFlexGroup
+        gutterSize="none"
+        direction="row"
+        responsive={false}
+        data-test-subj="dataTableCellActionPopoverTitle"
+      >
+        <EuiFlexItem>
+          <div style={{ maxWidth: '200px' }}>
+            <EuiText size="s">
+              <span className="unifiedDataTable__cellPopoverValue eui-textBreakWord">
```

Comment:
Referring to this "foreign" class name here seems brittle. Is there anything we can do to prevent accidental breakage? Maybe emulating the styles ourselves would be worth it?

Reply:
i placed myself in the shoes of the Unified Data Table developer who lets say introduces a change.

- If they change the CSS itself, then our will automatically be updated as we want uniform experience.
- If they decide to change the className itself, i would ideally do a findAll to see where it has been used and update accordingly.

I know the above does not cover all the edge cases, let's say they decide to add an additional class to the Popover content to get some additional style, then our won't get it. But in that case even if we copy paste the CSS, that would still mean the same.

Also as you mentioned in a later comment, we might move away from this component to UiActions, in that case this code is not going to live for long.

But yeah this is a brittle piece. Safest option is to style it ourselves. I will copy over the CSS 👍🏼

Reply:
There is also an option to introduce a wrapper component in UnifiedDataTable which includes these styles. Then you could import it in your code safely.